ouroboros-consensus/src-unix/Ouroboros/Consensus/Storage/IO.hs

```diff
@@ -82,6 +84,11 @@ write :: FHandle -> Ptr Word8 -> Int64 -> IO Word32
 write h data' bytes = withOpenHandle "write" h $ \fd ->
     fromIntegral <$> Posix.fdWriteBuf fd data' (fromIntegral bytes)
 
+pwrite :: FHandle -> Ptr Word8 -> Int64 -> Word64 -> IO Word32
```

Comment:
So the `Int64` is the number of bytes? Shouldn't that be a `Word64`?

I think you meant for the offset to be an `Int64`, because `FileOffset` is basically an `Int64`. Even then, what would a negative offset in a file mean? Like we did for `pread`, i would go for `Word64` (or `Word32` for the offset?) in both cases.

Reply:
I use Word64 for othe offset because AbsOffset is a Word64 wrapper. The number of bytes is Int64 because it's the same for write. But maybe this is a historical accident. The unix wrapper for write/pwrite gets a Word64 wrapper for the size, while the windows gets Word32. So I guess Word32 is the right choice for the size.